### PR TITLE
fix(repository): convert array items to json

### PR DESCRIPTION
**Pull request by raymondfeng**
A follow-up to #1711. This PR fixes the array conversion.

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: packages/repository/src/model.ts [hunk 1]
```diff
@@ -137,6 +137,32 @@ export class ModelDefinition {
...
+  }
+  // Handle arrays
+  if (Array.isArray(value)) {
+    return value.map(item => asJSON(item));
+  }
+  return value;
+}
+
+function asObject(value: any, options?: Options): any {
+  if (value == null) return value;
```
Comment (jannyHou) [inline packages/repository/src/model.ts]: Does `asObject` need check the flag for unknown properties as well like `toObject`?

`if (options && options.ignoreUnknownProperties === false) { // do the convertion}`

File: packages/repository/src/model.ts [hunk 2]
```diff
@@ -137,6 +137,32 @@ export class ModelDefinition {
   }
 }
 
+function asJSON(value: any): any {
+  if (value == null) return value;
```
Comment (virkt25) [inline packages/repository/src/model.ts]: Should there be a similar check for `undefined`?

File: packages/repository/src/model.ts (hunk 1 above)
Comment (raymondfeng) [reply packages/repository/src/model.ts]: No. The function is only used to handle 1st level properties.

File: packages/repository/src/model.ts (hunk 2 above)
Comment (raymondfeng) [reply packages/repository/src/model.ts]: Please note `== null` is `true` for both `undefined` and `null`.

**Pull request merged**